### DISCUSSION TITLE
进阶Q21不知道哪错了，一直80分

### DISCUSSION THREAD

**Discussion by luyao-pilgrim** [AmazingAng/WTF-Solidity#477] [Q&A] [answered]
1.代码复用
2.仅1正确
3.MyContract 需要 0xd9145CCE52D386f254917e481eB44e9943F39138 的某种许可，才可以调用其中的函数
4.(2)(4) 的返回结果分别是 10, 20
5.OtherContract(0xd9145CCE52D386f254917e481eB44e9943F39138).setX{value: 50}(x);

可以告诉我哪个错了，以及原因吗？万分感谢

**Comment by Rorke76753** [comment]
第二题

**Comment by JimHanss** [comment]
第二题错了，是(1)(2)都正确

**Reply by luyao-pilgrim** [reply]
依然还是80分，是什么情况

**Comment by luyao-pilgrim** [comment, marked as answer]
终于完成了
有两个错了
第二题：1，2都正确
第三题：安全隐患
